astronomer/providers/snowflake/hooks/snowflake.py

```diff
@@ -81,16 +83,28 @@ def run(  # type: ignore[override]
                 conn.commit()
         return self.query_ids
 
-    def check_query_output(self, query_ids: List[str]) -> None:
+    def check_query_output(
+        self, query_ids: List[str], handler: Callable | None = None, return_last: bool = True
+    ) -> Optional[Union[Any, List[Any]]]:
         """Once the query is finished fetch the result and log it in airflow"""
         with closing(self.get_conn()) as conn:
             with closing(conn.cursor(DictCursor)) as cur:
+                results = []
                 for query_id in query_ids:
                     cur.get_results_from_sfqid(query_id)
-                    cur.fetchall()
+                    if handler is not None:
+                        result = handler(cur)
+                        results.append(result)
```

Comment:
If handler is None, do we need to call `cur.fetchall()`? Or `cur.rowcount` in the below log line is accessible without calling `fetchall()` ?

Reply:
if handler is None by default, I am using `fetch_all_handler` which is `cur.fetchall()`